examples/persistable.php

```diff
@@ -0,0 +1,151 @@
+<?php
+declare(strict_types=1);
+
+namespace MongoDB\Examples;
+
+use MongoDB\BSON\ObjectId;
+use MongoDB\BSON\Persistable;
+use MongoDB\Client;
+use MongoDB\Model\BSONArray;
+use UnexpectedValueException;
+
+use function array_search;
+use function dirname;
+use function getenv;
+use function var_dump;
+
+require dirname(__FILE__) . '/../vendor/autoload.php';
+
+class PersistableEntry implements Persistable
+{
+    /** @var ObjectId */
+    private $id;
+
+    /** @var string */
+    private $name;
+
+    /** @var array<PersistableEmail> */
+    private $emails = [];
+
+    public function __construct(string $name)
+    {
+        $this->id = new ObjectId();
+        $this->name = $name;
+    }
+
+    public function getId(): ObjectId
+    {
+        return $this->id;
+    }
+
+    public function getName(): string
+    {
+        return $this->name;
+    }
+
+    public function setName(string $name): void
+    {
+        $this->name = $name;
+    }
+
+    public function getEmails(): array
+    {
+        return $this->emails;
+    }
+
+    public function addEmail(PersistableEmail $email): void
+    {
+        $this->emails[] = $email;
+    }
+
+    public function deleteEmail(PersistableEmail $email): void
+    {
+        $index = array_search($email, $this->emails, true);
```

Comment:
I noted that you used `$strict=true` here. Do you intend for this to only delete a specific PersistableEmail instance? This method doesn't appear to be used in the example script so it's hard to say what the intended use case is.

At a glance (and before I realized PersistableEmail was a document), I was going to point out that `addEmail` and `deleteEmail` seemed inconsistent, since `addEmail` appends unconditionally (and could add duplicates).

Reply:
Yeah, I attempted to make this look like a business object. I think I'll go back and make the properties public and remove the accessors.